### ISSUE TITLE
Incestuous genealogical trees

### ISSUE THREAD

**Issue by CuloArdido** [zhmylove/genealogy#1]
Any ideas on how to make family trees of incestuous families? Because I have been working on [this](http://www.plantuml.com/plantuml/uml/VLRlRjis4F-kfpWK0ve4I68dhjS2MT3DQxUs0Hmq7P1s6ugpHDjCgADBKblTOK0VP7kvFib8wKJmXc7w8E3tXtVadOxsKn_GXQurbR_Nj4E7BRIMRBrrjbKGNASAfL5hx4nuepw6RutUKAieG1rVoXKkf-g0j34A_gW0dEseMLX7oi7Pe-hFgWewHEqvjjhiuOrJ2YvySFfUNI8ry89gvKDdEtznoZHSsQuvlT44htJZg-hXk7YhiF5mi75nOSojA7X8CNPE_QPZ1DWeKWw3jkIXHW8qti9AWQRQT8rg3a-gl5juSBKdwdF1ZJlvXUQJ1u0UJ6HBrsiTi7Sb1nmq2fl3tkYYa3QtT7gZ2Gs7IhaFhc96QwZVtLPxyDe7vVnX6PWz7FlyoP1z88NFEiscsQF151dXK9GTiy6XI3ewmj5nFZWY8nr8e-EC7KZ_NQwNgLopQIxNdPEHwlrugAy7JsJaYGoE3ZvHP8G3wTyE9EdGFZjulv6H3fFtmQsDJEhOOgjDunI1NKFPAQy9xdNZPVjfhClPq28N6NaQp0wkYKNfq2eVPyfzJfIihiiMhbfaP9JLVdHmboDJEhv7HoWjJQ9aTtRPClPv9nsgpNcuRf6H3etiUCm6BKDWlGUP93BoC9z61_St4dAVfcC35pKIic3nBRSPcNuWmrOvqSIRe8n1UBV5Ba-QxpdHQhjIBbZuylclx-ngjV_o-M-uGGeQDpRFtYefOfNIX6dDSwVlhD41vJekW4ks_yiMfwfS624VZ7FUkNIsVWXlNF-vfflfeGUxNlkTqxIfJ1AFZoyXtoBHKHA5mphDCgD2KyGQerBh25bZL6h8AJFA6byJh6PShAtth38gzsgGDKQ5XiGQerBJdZL6GZDbFhuO-3AGHHsutXS9Fc56XJpCVQfbHgNCCv_gMK1XI1EVQfgHbDEufx9cLCeywiDdNH25PHZrLDoCf9pcdFCpahB6AfK_8oaZD-tEPImDQSHJYpAISXhmrAK2YU8ObOkNP_srCiFl_LRlVDKF2fmulTc64_wgEGIJFFPfwN9iSMmss7HdW4lHL0FhuVz6gzeuenmsfUKfJe4BcEKygDcG9pTb2Wg-GkVix-yP_Nm-_pYTdiu-_GhB_MGwdzQF9pDQ_7Xs-_QwldxxPDw-MBZ9zMovdTo-cxsS8itgsRFRdpvCvw-ctR6TBNDSlpCwl3UQL1cSO_iVvfCFo_bYSVviqbQnY1j53HYr3XN0qT5LZu-E8lWcFXN0XTZd9SGdLol9Cbtso6R4c4EcgdeQtrrh_W40) but I would like to redo it with your extension. In case it is not contemplated, the following image could be used as an example where the consanguinity between parents is indicated:
![Example of a possible incestuous family tree](https://cpb-eu-w2.wpmucdn.com/blogs.ucl.ac.uk/dist/9/186/files/2018/08/Pic4.png)
When trying to use genealogy.iuml, the following doubts also arose:
1. Can strings be used instead of numbers as identifiers when making a `person` instance?
2. How do I get the `pair_id` that can be used in the `pair`, `doublepair` and `child` procedures?
3. In example.puml the tree is generated from bottom to top, is that normal?
4. How can I hide the nodes so that the descendants come directly from the lines as in [this case](https://external-preview.redd.it/AtbZeHEP8fGs_t0qcihg6WTEzRUZWBW6Ex2b5_0BZv0.jpg?auto=webp&s=e4b482ff392f654f87ebd4df277bb2fa5045722c)?
5. Is it possible to change the style and color of some lines/joins?

**Comment by zhmylove** [comment]
Hi @CuloArdido 
Thank you for paying attention to this module.
Currently I'm on a vacation and probably I'll be ready to improve the project since 10th Aug as I don't have an access to my PC now.

As for incestuous families, I've checked that there is no problem when the bond is between brother and sister (like on yours picture). But when the parental bond is between parent/child, then plantuml fails to place some elements. Although the edges are correct. In any case, just child() addition works as expected.

I haven't tested if strings can be used instead of numbers. You can try it out. I bet this will do.

How do I get the pair_id...
You just select them by your own. Just see this inside example: I've picked them just sequentially, but they could be any.

...the tree is generated from bottom to top...
I believe, you can write them in any order. Plantuml will sort them out correctly.

Is it possible to change the style and color of some lines/joins?
Absolutely yes, just like in regular plantuml/graphviz.

How can I hide the nodes so that the descendants come directly from the lines...
I don't think, we can hide "interface" objects somehow.
I was just thinking about re-implementing the whole thing on a class diagram, instead of on a component. It'll take some time to investigate around it, so I'll be ready to continue only after vacation...
Feel free to modify the source in any ways, I'll be glad to get pull requests. 
Thank you!

**Comment by zhmylove** [comment]
@CuloArdido it seems, class diagram lacks good positioning:
```
@startuml
together {
class Child2
class Child1
}
together {
class Parent2
class Parent1
}
Parent1 - Parent2
(Parent1, Parent2) -- Child1
(Parent1, Parent2) -- Child2
@enduml
```

**Comment by zhmylove** [comment]
Even without grouping, compare:
```
@startuml
class Child2
class Child1
class Parent2
class Parent1
Parent1 - Parent2
(Parent1, Parent2) -- Child1
(Parent1, Parent2) -- Child2
@enduml
```
with
```
@startuml
class Child2
class Child1
class Parent2
class Parent1
Parent1 - Parent2
(Parent1, Parent2) -- Child1
@enduml
```

**Comment by CuloArdido** [comment]
>You just select them by your own. Just see this inside example

Is that by the way the example was written I thought that those were shared identifiers or something (anyway I was modifying it to make it clearer how it works).
>you can write them in any order. Plantuml will sort them out correctly.

I was talking about this:
![example.uyml generated diagram](https://user-images.githubusercontent.com/20852424/128618527-2ee08165-53a4-4b93-8265-711e830d30d7.png)
I found it strange that it didn't start with Person 4 and Person 5 at the top (being the first generation) so that it was then going down with the offspring of Person 4.
>I don't think, we can hide "interface" objects somehow.
>I was just thinking about re-implementing the whole thing on a class diagram.

Maybe `split` could be used as shown [here](https://plantuml.com/activity-diagram-beta) to make the unions without the interface, but I have no idea how to adapt it to this type of diagrams. Another option might be [WBS](https://plantuml.com/es/wbs-diagram) but I'm not sure if it would work well when incest is involved.

**Comment by zhmylove** [comment]
I've merged #2, thank you!

As for WBS and Activity+split, they won't do, I believe :(
It's due to they still try to position elements relative to single root.
